### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by pi8027**

### CODE REVIEW

File: meta.yml [hunk 1]
```diff
@@ -87,10 +85,140 @@ dependencies:
...
+  Although the `ring` tactic supports ring homomorphisms (explained below), all
+  the monomials and polynomials `m_1, .., m_n, p_1, .., p_n, p, q` must have the
+  same type `R` for the moment.
+
+  Each tactic provided by this library has a preprocessor and supports
+  applications of (semi)ring homomorphisms and additive functions (N-module or
+  Z-module homomorphisms).
+  For example, suppose `f : S -> T` and `g : R -> S` are ring homomorphisms. The
+  preprocessor turns a ring sub-expression of the form `f (x + g (y * z))` into
+  `f x + f (g y) + f (g z)`.
```
Comment (proux01) [inline meta.yml:126]:
```suggestion
  `f x + f (g y) * f (g z)`.
```

File: meta.yml [hunk 2]
```diff
@@ -87,10 +85,140 @@ dependencies:
...
+  `f x + y%:~R * z%:~R` where `f \o g : int -> T` is replaced with `intr`
+  (`_%:~R`).
+  Thanks to the preprocessor, the `ring` tactic supports the following
+  constructs apart from homomorphism applications:
+  - `GRing.zero` (`0%R`),
+  - `GRing.add` (`+%R`),
+  - `addn`,
+  - `N.add`,
+  - `Z.add`,
+  - `GRing.natmul`,
```
Comment (proux01) [inline meta.yml:139]:
```suggestion
  - `GRing.natmul` (`_%:R`),
```

File: meta.yml [hunk 3]
```diff
@@ -87,10 +85,140 @@ dependencies:
...
+  - `GRing.opp` (`-%R`),
+  - `Z.opp`,
+  - `Z.sub`,
+  - `intmul`,
```
Comment (proux01) [inline meta.yml:143]:
```suggestion
  - `intmul` (`_%:~R`),
```

File: meta.yml [hunk 4]
```diff
@@ -87,10 +85,140 @@ dependencies:
...
+  - `GRing.mul` (`*%R`),
+  - `muln`,
+  - `N.mul`,
+  - `Z.mul`,
+  - `GRing.exp`,[^constant_exponent]
+  - `exprz`,[^constant_exponent]
+  - `expn`,[^constant_exponent]
+  - `N.pow`,[^constant_exponent]
+  - `Z.pow`,[^constant_exponent]
+  - `GRing.inv`,
```
Comment (proux01) [inline meta.yml:154]:
```suggestion
  - `GRing.inv` (`_^-1`, used in notation `x / y` as `x * y^-1`),
```


but isn't it only for the `field` tactic?

File: meta.yml [hunk 5]
```diff
@@ -87,10 +85,140 @@ dependencies:
...
+  (`_%:~R`).
+  Thanks to the preprocessor, the `ring` tactic supports the following
+  constructs apart from homomorphism applications:
+  - `GRing.zero` (`0%R`),
+  - `GRing.add` (`+%R`),
+  - `addn`,
+  - `N.add`,
+  - `Z.add`,
+  - `GRing.natmul`,
+  - `GRing.opp` (`-%R`),
```
Comment (proux01) [inline meta.yml:140]:
```suggestion
  - `GRing.opp` (`-%R`, used in notation `x - y` as `x + (- y)`),
```

File: meta.yml [hunk 6]
```diff
@@ -87,10 +85,140 @@ dependencies:
...
+  - `GRing.exp`,[^constant_exponent]
+  - `exprz`,[^constant_exponent]
+  - `expn`,[^constant_exponent]
+  - `N.pow`,[^constant_exponent]
+  - `Z.pow`,[^constant_exponent]
+  - `GRing.inv`,
+  - `S`,
+  - `Posz`,
+  - `Negz`, and
+  - constants of type `nat`, `N`, or `Z`.
```
Comment (proux01) [inline meta.yml:158]:
```suggestion
  - constants of type `nat`, `int`, `N`, or `Z`.
```
(IINM)

File: meta.yml (hunk 2 above)
Comment (pi8027) [reply meta.yml:139]: I expect the users to learn this kind of notation from the header document of ssralg.

File: meta.yml (hunk 4 above)
Comment (pi8027) [reply meta.yml:154]: You are right. I should remove this line.

File: meta.yml (hunk 6 above)
Comment (pi8027) [reply meta.yml:158]: An `int` constant is always either `Posz` or `Negz` applied to a `nat` constant, and the preprocessor supports these ingredients separately.

**Pull request merged**